### PR TITLE
Resolves TimerTriggerVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the TimerTriggerVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/Trigger/TimerTriggerVisualizerSystem.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                sound.KeyFrames.Add(new AnimationTrackPlaySound.KeyFrame(comp.CountdownSound.GetSound(), 0));
+            }
+        }
+    }
+
+    protected override void OnAppearanceChange(EntityUid uid, TimerTriggerVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+        if(!TryComp<AnimationPlayerComponent>(uid, out var animPlayer))
```
Comment (AJCM-git) [inline Content.Client/Trigger/TimerTriggerVisualizerSystem.cs:37]: newline

File: Content.Client/Trigger/TimerTriggerVisualizerSystem.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            flick.LayerKey = TriggerVisualLayers.Base;
+            flick.KeyFrames.Add(new AnimationTrackSpriteFlick.KeyFrame("primed", 0f));
+
+            if (comp.CountdownSound != null)
+            {
+                var sound = new AnimationTrackPlaySound();
+                comp.PrimingAnimation.AnimationTracks.Add(sound);
+                sound.KeyFrames.Add(new AnimationTrackPlaySound.KeyFrame(comp.CountdownSound.GetSound(), 0));
+            }
+        }
```
Comment (AJCM-git) [inline Content.Client/Trigger/TimerTriggerVisualizerSystem.cs:30]: single object initializer, unhardcode

File: Content.Client/Trigger/TimerTriggerVisualizerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Trigger;
+
+public sealed class TimerTriggerVisualizerSystem : VisualizerSystem<TimerTriggerVisualizerComponent>
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<TimerTriggerVisualizerComponent, ComponentInit>(OnInit);
```
Comment (AJCM-git) [inline Content.Client/Trigger/TimerTriggerVisualizerSystem.cs:12]: OnComponentInit

File: Content.Client/Trigger/TimerTriggerVisualizerSystem.cs (hunk 1 above)
Comment (AJCM-git) [inline Content.Client/Trigger/TimerTriggerVisualizerSystem.cs:37]: newline

File: Content.Client/Trigger/TimerTriggerVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    protected override void OnAppearanceChange(EntityUid uid, TimerTriggerVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+        if(!TryComp<AnimationPlayerComponent>(uid, out var animPlayer))
+            return;
+
+        if (!AppearanceSystem.TryGetData(uid, TriggerVisuals.VisualState, out TriggerVisualState state, args.Component))
```
Comment (AJCM-git) [inline Content.Client/Trigger/TimerTriggerVisualizerSystem.cs:40]: generic

File: Content.Client/Trigger/TimerTriggerVisualizerSystem.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        switch (state)
+        {
+            case TriggerVisualState.Primed:
+                if (!AnimationSystem.HasRunningAnimation(uid, animPlayer, TimerTriggerVisualizerComponent.AnimationKey))
+                {
+                    AnimationSystem.Play(uid, animPlayer, comp.PrimingAnimation, TimerTriggerVisualizerComponent.AnimationKey);
+                }
+                break;
+            case TriggerVisualState.Unprimed:
+                args.Sprite.LayerSetState(0, "icon");
```
Comment (AJCM-git) [inline Content.Client/Trigger/TimerTriggerVisualizerSystem.cs:54]: unhardcode

File: Content.Client/Trigger/TimerTriggerVisualizerComponent.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+using Robust.Client.Animations;
+using Robust.Shared.Audio;
+
+namespace Content.Client.Trigger;
+
+[RegisterComponent]
+[Access(typeof(TimerTriggerVisualizerSystem))]
+public sealed class TimerTriggerVisualizerComponent : Component
```
Comment (AJCM-git) [inline Content.Client/Trigger/TimerTriggerVisualizerComponent.cs:8]: rename to visuals

File: Content.Server/Entry/IgnoredComponents.cs [hunk 7]
```diff
@@ -20,6 +20,7 @@ public static class IgnoredComponents
             "PuddleVisualizer",
             "UIFragment",
             "PDABorderColor",
+            "TimerTriggerVisualizer",
```
Comment (AJCM-git) [inline Content.Server/Entry/IgnoredComponents.cs:23]: remove

File: Content.Client/Trigger/TimerTriggerVisualizerComponent.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public const string AnimationKey = "priming_animation";
+
+    [DataField("countdown_sound")]
```
Comment (AJCM-git) [inline Content.Client/Trigger/TimerTriggerVisualizerComponent.cs:12]: camelCase, vv

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Comment by DrSmugleaf** [comment]
Reopen when you come back to this.

**Comment by TemporalOroboros** [comment]
@DrSmugleaf 
Commits tested and pushed, please reopen whenever is most convenient.

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**